objects/src/notes/inputs.rs

```diff
@@ -79,6 +79,20 @@ impl NoteInputs {
     pub fn to_vec(&self) -> Vec<Felt> {
         self.values.to_vec()
     }
+
+    // UTILITIES
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the key under which the raw (un-padded inputs) are located in the advice map.
+    ///
+    /// TODO: eventually, this should go away. for now we need it because note inputs for input
+    /// notes are padded, while note inputs for expected output notes are note. switching to a
+    /// different padding scheme (e.g., RPX) would eliminate the need to have two different keys.
+    pub fn commitment_to_key(commitment: Digest) -> Digest {
+        let mut key: Word = commitment.into();
+        key[3] += ONE;
```

Comment:
I'm guessing this cannot overflow in any way

Reply:
Yep - it's a field element addition. Won't overflow.

Reply:
Or rather it'll just wrap around.